processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -681,7 +675,8 @@ private Assignment forgeMapMapping(Type sourceType, Type targetType, SourceRHS s
         private String getName(Type sourceType, Type targetType) {
             String fromName = getName( sourceType );
             String toName = getName( targetType );
-            return Strings.decapitalize( fromName + "To" + toName );
+            String with = nullValueMappingStrategy != null ? "With" + nullValueMappingStrategy.name() : "";
```

Comment:
The generated name follows the enum constant. So capitals and underscore.. Should I decapitilize?